### PR TITLE
Fix/8122 8127 profile page telegram sub

### PR DESCRIPTION
**Pull request by DenBlaz**
Changed the logic of the switcher
Added the logic of comparing the switcher status with the saved status
Added tests
Fixed the 404 error
Added the selection confirmation window
Edited the page logic for the correct operation of the notification subscription functionality
Edited the logic of the Save changes button

## Summary by CodeRabbit

* **New Features**
  * Telegram notification switch now uses event-driven toggles with optimistic save, rollback on error, and persisted saved-state syncing. Confirmation flow added for navigating away when changing this setting.

* **Bug Fixes**
  * Save/Cancel buttons no longer submit forms; Cancel restores server state. Submit enabling tightened; form dirty/pristine handling, phone/email normalization, and post-save UI/state sync improved.

* **Tests**
  * Expanded coverage for switch behavior, telegram flows, bot URL handling, submission/cancellation, and numerous edge cases.

### CODE REVIEW

File: src/app/ubs/ubs-user/components/ubs-user-profile-page/ubs-user-profile-page.component.ts [hunk 1]
```diff
@@ -209,30 +217,59 @@ export class UbsUserProfilePageComponent implements OnInit, OnDestroy {
...
   onCancel(): void {
     this.userProfile.addressDto = [...this.savedUserAddresses];
+    this.userProfile.telegramIsNotify = this.savedTelegramIsNotify;
     this.tempAddedAddressHolder.length = 0;
     this.tempRemovedAddressHolder.length = 0;
-    this.userInit();
-    this.isEditing = false;
+    this.clientProfileService
+      .getDataClientProfile()
+      .pipe(take(1))
```
Comment (witolDark) [inline src/app/ubs/ubs-user/components/ubs-user-profile-page/ubs-user-profile-page.component.ts]: Take(1) is redundant for http.get

File: src/app/ubs/ubs-user/components/ubs-user-profile-page/ubs-user-profile-page.component.ts [hunk 2]
```diff
@@ -184,7 +181,18 @@ export class UbsUserProfilePageComponent implements OnInit, OnDestroy {
...
   }
 
-  isSubmitBtnDisabled() {
+  isSubmitBtnDisabled(): boolean {
+    if (!this.userForm) {
+      return true;
+    }
+
+    const formSwitch = !!this.userForm.get('telegramIsNotify')?.value;
+    const serverSwitch = !!this.savedTelegramIsNotify;
```
Comment (witolDark) [inline src/app/ubs/ubs-user/components/ubs-user-profile-page/ubs-user-profile-page.component.ts]: !! is applied to boolean variable

File: src/app/ubs/ubs-user/components/ubs-user-profile-page/ubs-user-profile-page.component.ts [hunk 3]
```diff
@@ -442,32 +497,72 @@ export class UbsUserProfilePageComponent implements OnInit, OnDestroy {
...
         .pipe(take(1))
         .subscribe((confirmed) => {
           if (confirmed) {
             this.userProfile.telegramIsNotify = true;
-            this.userForm.markAsDirty();
-            this.userForm.get('telegramIsNotify')?.setValue(true);
+            this.userForm.get('telegramIsNotify')?.setValue(true, { emitEvent: false });
+
+            if (this.isEditing) {
+              if (this.savedTelegramIsNotify !== true) {
```
Comment (witolDark) [inline src/app/ubs/ubs-user/components/ubs-user-profile-page/ubs-user-profile-page.component.ts]: !this.savedTelegramIsNotify

**Comment by jaworovcki** [comment]
Погано!
